lib/pure/random.nim

```diff
@@ -163,6 +163,27 @@ proc rand*[T](a: openArray[T]): T =
   ## returns a random element from the openarray `a`.
   result = a[rand(a.low..a.high)]
 
+import algorithm                    #For upperBound
+proc rand*[T, U](r: var Rand; a: openArray[T], w: openArray[U], n=1): seq[T] =
```

Comment:
I would call this ``choose`` or something other than ``rand``, we've got too many "rand"s already.

Reply:
I originally called it `sample`, but then saw there was already a `proc rand*[T](a: openArray[T]): T` (directly above it in my PR), and this new proc is conceptually a generalization of that existing one from uniform weights to uneven weights.

Whatever ident we decide, the uniform/equal-weighted `(a: openArray[T])` variant on sampling from an openArray should probably be the same name.  If we are renaming the old we probably need a deprecation, etc.

Reply:
(There's also an older deprecated `proc random*[T](a: openArray[T]): T`.)

Reply:
Yeah, let's rename the other one too. I'm happy with ``sample``.